src/libraries/System.Numerics.Tensors/ref/System.Numerics.Tensors.netcore.cs

```diff
@@ -4,8 +4,267 @@
+        System.Numerics.Tensors.TensorSpan<T> AsTensorSpan(scoped System.ReadOnlySpan<nint> start);
+        void Clear();
+        static abstract TSelf Create(System.ReadOnlySpan<nint> lengths, bool pinned = false);
+        static abstract TSelf Create(System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides, bool pinned = false);
+        static abstract TSelf CreateUninitialized(System.ReadOnlySpan<nint> lengths, bool pinned = false);
+        static abstract TSelf CreateUninitialized(System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides, bool pinned = false);
+        void Fill(T value);
+        new ref T GetPinnableReference();
+    }
+    public readonly ref partial struct ReadOnlyTensorSpan<T>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public ReadOnlyTensorSpan(System.Span<T> span) { throw null; }
+        public ReadOnlyTensorSpan(System.Span<T> span, scoped System.ReadOnlySpan<nint> lengths, scoped System.ReadOnlySpan<nint> strides) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public unsafe ReadOnlyTensorSpan(T* data, nint dataLength) { throw null; }
+        [System.CLSCompliantAttribute(false)]
+        public unsafe ReadOnlyTensorSpan(T* data, nint dataLength, scoped System.ReadOnlySpan<nint> lengths, scoped System.ReadOnlySpan<nint> strides) { throw null; }
+        public ReadOnlyTensorSpan(T[]? array) { throw null; }
+        public ReadOnlyTensorSpan(T[]? array, System.Index startIndex, System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides) { throw null; }
+        public ReadOnlyTensorSpan(T[]? array, int start, System.ReadOnlySpan<nint> lengths, System.ReadOnlySpan<nint> strides) { throw null; }
+        public static System.Numerics.Tensors.ReadOnlyTensorSpan<T> Empty { get { throw null; } }
+        public nint FlattenedLength { get { throw null; } }
+        public bool IsEmpty { get { throw null; } }
+        public ref readonly T this[scoped System.ReadOnlySpan<System.Buffers.NIndex> indexes] { get { throw null; } }
+        public System.Numerics.Tensors.ReadOnlyTensorSpan<T> this[scoped System.ReadOnlySpan<System.Buffers.NRange> ranges] { get { throw null; } }
+        public ref readonly T this[scoped System.ReadOnlySpan<nint> indexes] { get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.UnscopedRefAttribute]
+        public System.ReadOnlySpan<nint> Lengths { get { throw null; } }
+        public int Rank { get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.UnscopedRefAttribute]
+        public System.ReadOnlySpan<nint> Strides { get { throw null; } }
+        public static System.Numerics.Tensors.ReadOnlyTensorSpan<T> CastUp<TDerived>(System.Numerics.Tensors.ReadOnlyTensorSpan<TDerived> items) where TDerived : class?, T? { throw null; }
+        public void CopyTo(System.Numerics.Tensors.TensorSpan<T> destination) { }
```

Comment:
This and other span inputs on the ref structs should be `scoped` to indicate that they will never be captured by the span itself.